src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -79,6 +80,17 @@
    jruby-request/wrap-with-error-handling
    ring/wrap-params))
 
+(defn static-file-content-request-handler
+  "Build the main request handler for the /static_file_content endpoint"
+  [get-code-content]
+  (fn [req]
+    (let [environment (get-in req [:params "environment"])
+          code-id (get-in req [:params "code-id"])
+          file-path (get-in req [:params :rest])
+          content-stream (get-code-content environment code-id file-path)]
+      {:status 200
```

Comment:
On further thought ...

Maybe this code should simply be responsible for checking that environment and code ID are not `nil`, and that `file-path` is actually a file which exists on disk?  Although, for that last case, this code can't easily know that, since it's actually inside an environment.  This is more complex than I had originally realized.